### PR TITLE
Python 3.x compatible (error on OAuth)

### PR DESCRIPTION
**Pull request by jgsogo**
These are some changes to make this package compatible with Python 3.

I've only tested *Import API*. In order to work with *OAuth*, [python-oauth2](https://github.com/joestump/python-oauth2) must be upgraded to py3 first.

### CODE REVIEW

**Comment by jgsogo** [comment]
With the release 1.9 of `python-oauth2` (now in Pypi), CargoDBOAuth works with python3 too, nevertheless it raises an exception when calling CartoDB servers (any docs about this?):

```python
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "E:\_VirtualEnvs\adif-scraper\lib\site-packages\django\core\management\__init__.py", line 338, in execute_from_command_line
    utility.execute()
  File "E:\_VirtualEnvs\adif-scraper\lib\site-packages\django\core\management\__init__.py", line 330, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "E:\_VirtualEnvs\adif-scraper\lib\site-packages\django\core\management\base.py", line 393, in run_from_argv
    self.execute(*args, **cmd_options)
  File "E:\_VirtualEnvs\adif-scraper\lib\site-packages\django\core\management\base.py", line 444, in execute
    output = self.handle(*args, **options)
  File "E:\_VirtualEnvs\adif-scraper\adif-scraper\adif\management\commands\dump_stations.py", line 65, in handle
    success = upload_to_cartodb(filename, CARTODB_API_KEY, CARTODB_DOMAIN)
  File "E:\_VirtualEnvs\adif-scraper\adif-scraper\adif\utils\cartodb.py", line 24, in upload_to_cartodb
    fi.run()
  File "E:\_VirtualEnvs\adif-scraper\lib\site-packages\cartodb-0.8.1-py3.4.egg\cartodb\imports.py", line 88, in run
  File "E:\_VirtualEnvs\adif-scraper\lib\site-packages\cartodb-0.8.1-py3.4.egg\cartodb\imports.py", line 49, in req
  File "E:\_VirtualEnvs\adif-scraper\lib\site-packages\cartodb-0.8.1-py3.4.egg\cartodb\cartodb.py", line 146, in get_response_data
cartodb.cartodb.CartoDBException: {'imports': 'invalid user max_concurrent_import_count (must be positive integer)'}
```

**Comment by danicarrion** [comment]
@jgsogo this looks like a platform problem. can you confirm this is still happening?

also, before accepting your pull request, you need to sign our contributor agreement https://cartodb.com/contributing/ i'm really sorry for the inconvenience

**Comment by jgsogo** [comment]
I've just sent the email for the CLA ;D

About the `max_concurrent_import_count` error, it is still raised. It seems to be related to source code in [this file](https://github.com/CartoDB/cartodb/blob/master/services/platform-limits/lib/importer/user_concurrent_imports_amount.rb) from CartoDB/cartodb repo (it is the only search result in Google).  May I open an issue there or do you think it is originated somewhere else?

**Comment by danicarrion** [comment]
what's the name of the cartodb account you're using?

**Comment by jgsogo** [comment]
This is the config data I'm using:

```
    user =  'jgsogo@gmail.com'
    password =  '<password>'
    CONSUMER_KEY='<consumer-secret-key>'
    CONSUMER_SECRET='<consumer-secret>'
    cartodb_domain = 'jgsogo'
    cl = CartoDBOAuth(CONSUMER_KEY, CONSUMER_SECRET, user, password, cartodb_domain)
```

**Comment by danicarrion** [comment]
Our Import API has a broken xAuth support, that's why we don't officially claim to support xAuth there. Can you try with your API key?

**Comment by jgsogo** [comment]
The ImportAPI works ok (commits on 3rd Sept.), and is the one I'm using; but now that oauth2 has been ported to Python3 (oauth2>=1.9) I also make the neccesary changes to this package to "work" with it.

**Comment by jgsogo** [comment]
CLA signed and sent by email, may I/you wait for confirmation?

**Comment by vhamer** [comment]
Thank you Javier! We have received your CLA. I am reassigning to @danicarrion for next steps. Thanks for your contribution.

File: cartodb/cartodb.py [hunk 1]
```diff
@@ -138,7 +143,7 @@ def get_response_data(self, resp, parse_json=True):
                 return resp.json()
             return resp.content
         elif resp.status_code == requests.codes.bad_request:
-            raise CartoDBException(resp.json()['error'])
+            raise CartoDBException(resp.json()['errors'])
```
Comment (danicarrion) [inline cartodb/cartodb.py]: Why have you changed this? 'error' is correct, if I'm not mistaken (you can run the tests to verify)

**Comment by danicarrion** [comment]
With python3 I get some warnings about the import file & sockets not being properly closed. I'll issue a ticket and fix it after we merge your PR.

File: cartodb/cartodb.py (hunk 1 above)
Comment (jgsogo) [inline cartodb/cartodb.py]: Interesting... In fact when you run the test you get ```error``` in (for example):

```
{'error': ['relation "my_table_with_some_data" does not exist']}
```

but the error related to ```max_concurrent_import_count``` (when everything should work as expected) is:

```
{'errors': {'imports': 'invalid user max_concurrent_import_count (must be positive integer)'}}
```

I think that this should be standarized in the platform, don't you? Maybe we can make a list with the content of both ```error|s``` and report an issue upstreams.

Comment (danicarrion) [inline cartodb/cartodb.py]: Sorry I somehow missed the notification for this. Could you simply try one and if KeyError is raised then go for the other option?

Comment (jgsogo) [inline cartodb/cartodb.py]: Sure!

**Comment by jgsogo** [comment]
I like it this way:

1. Raise ```error``` if available (it is the expected behaviour)
2. Raise ```errors``` ("not expected", we don't know how to solve it)
3. Otherwise raise ```response.text``` to give information to the user.

**Comment by danicarrion** [comment]
thanks!!!

**Pull request merged**